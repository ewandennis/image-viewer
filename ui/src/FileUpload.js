import React, { useCallback } from 'react';
import { DropzoneArea } from "material-ui-dropzone";
import { maxFileSizeBytes } from "./config";

export default function FileUpload({ onUploadFile }) {
  const handleChange = useCallback(
    fileList => {
      if (fileList && fileList.length > 0) {
        onUploadFile(fileList[0]);
      }
    },
    [onUploadFile]
  );

  return (
    <DropzoneArea
      acceptedFiles={["image/*"]}
      filesLimit={1}
      maxFileSize={maxFileSizeBytes}
      showAlerts={false}
      showPreviews={false}
      showPreviewsInDropzone={false}
      onChange={handleChange}
    />
  );
}
