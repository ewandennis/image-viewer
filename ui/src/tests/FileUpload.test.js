import React from 'react';
import { shallow } from 'enzyme';
import FileUpload from '../FileUpload';

describe('FileUpload', () => {
  const subject = (props = {}) => shallow(<FileUpload {...props} />);

  it('renders', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('calls onUploadFile', () => {
    const onUploadFileMock = jest.fn();
    const wrapper = subject({ onUploadFile: onUploadFileMock });
    const file = { name: 'iamafile.png' };
    wrapper.prop('onChange')([ file ]);
    expect(onUploadFileMock).toHaveBeenCalledTimes(1);
    expect(onUploadFileMock).toHaveBeenCalledWith(file);
  });
});
