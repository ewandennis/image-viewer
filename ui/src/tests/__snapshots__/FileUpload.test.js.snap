// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload renders 1`] = `
<WithStyles(DropzoneArea)
  acceptedFiles={
    Array [
      "image/*",
    ]
  }
  filesLimit={1}
  maxFileSize={1000000}
  onChange={[Function]}
  showAlerts={false}
  showPreviews={false}
  showPreviewsInDropzone={false}
/>
`;
